bluetooth/src/commonMain/kotlin/scanner/Scanner.kt

```diff
@@ -92,22 +92,18 @@ abstract class BaseScanner constructor(
         val autoRequestPermission: Boolean = true,
         val autoEnableSensors: Boolean = true,
         val eventBufferSize: Int = DEFAULT_EVENT_BUFFER_SIZE,
-        val logLevel: LogLevel = LogLevel.NONE
+        val logLevel: RestrictedLogLevel = RestrictedLogLevel.None
     )
 
-    enum class LogLevel {
-        NONE,
-        INFO,
-        VERBOSE
-    }
-
     interface Builder {
         fun create(
             settings: Settings,
             coroutineScope: CoroutineScope
         ): BaseScanner
     }
 
+    private val logger = RestrictedLogger(settings.logLevel)
```

Comment:
It's better to inject logger in the constructor (once it's interface, see comment below) so we can use different one (like stub in unit tests or log file writer)